BUILD.md

```diff
@@ -0,0 +1,212 @@
+```bash
+$ sudo systemctl daemon-reload
+$ sudo systemctl restart docker
+```
+
+Verify that the configuration has been loaded:
+```bash
+$ systemctl show --property=Environment docker
+Environment=HTTP_PROXY=http://proxy.example.com:80/
+```
+
+Step Two: Clone The Repository
+-------------
+
+You'll need to have git installed in order to clone the Sawtooth source
+code repository. You can find up-to-date installation instructions
+at [Getting Started - Installing Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git).
+
+Open up a terminal and run the following:
+
+```bash
+   $ cd $HOME
+   $ mkdir sawtooth
+   $ cd sawtooth
+   $ git clone https://github.com/hyperledger/sawtooth-core.git
+```
+
+Step Three: Initializing Your Environment
+-------------
+
+Some of the components of Sawtooth depend partially on code that must first be
+built. This includes generating protobuf classes for each language. To build
+all of the components of Sawtooth, run this command:
+
+```bash
+  $ bin/build_all
+```
+
+By default this builds all the components for all the components and sdks in
+the repository. This is a very slow process. It is recommended that you only
+build for the language dependencies you need for your development purposes. The
+minimum requirements for running the Sawtooth validator is the python build:
+
+```bash
+  $ bin/build_all -l python
+```
+
+To get details on all the options for the build_all script
+
+```bash
+  $ bin/build_all -h
+```
+
+If you are working on the core validate the ‘python’ language is all that is
+required. If you are working on a particular language sdk then you will need to
+build the language that sdk is for as well.
+
+**Note:** This build environment uses docker to virtualize the build and
+execute the code in the development directory. This allows for modification
+made to the local source to be built and tested without installing local
+dependencies on your machine.
+
+**Note:**  If you wish to configure your development machine to do compilation
+directly on the host without a docker virtualization for the build
+environment, the docker files in the sawtooth-core/docker. For example
+sawtooth-core/docker/sawtooth-dev-python describes the configuration and
+components needed to build and run the python components in the system.
+
+Step Four: Development Tasks
+-------------
+
+To run a full Validator Node from the local source:
+
+```bash
+  $ docker-compose -f docker/compose/sawtooth-local.yaml up
+```
+
+This will start a validator with the following components attached to it:
+  - Xo Transaction Processor
+  - IntKey Transaction Processor
+  - Settings Transaction Processor
+  - REST API - available on host port 8008
+  - Shell - for running sawtooth cli commands
+
+From another console window you can access the shell
+```bash
+  $ docker-compose -f docker/compose/sawtooth-local.yaml exec shell bash
+```
+
+This command uses docker compose and the development docker images. These
```

Comment:
used in this way, Docker Compose is a proper name (vs, `docker-compose` the cli command itself)